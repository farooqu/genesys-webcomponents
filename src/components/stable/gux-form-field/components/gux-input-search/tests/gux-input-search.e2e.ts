import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../../tests/e2eTestUtils';

describe('gux-input-search', () => {
  describe('#render', () => {
    [
      `<gux-input-search lang="en">
        <input name="search" type="search" placeholder="Enter a search" />
      </gux-input-search>`,
      `<gux-input-search lang="en">
        <input name="search" type="search" placeholder="Enter a search" value="test" />
      </gux-input-search>`,
      `<gux-input-search lang="en">
        <input name="search" type="search" placeholder="Enter a search" disabled/>
      </gux-input-search>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-input-search');
        await a11yCheck(page);

        expect(element.innerHTML).toMatchSnapshot();
      });
    });
  });

  describe('clear button', () => {
    it('should appear when the value is not empty', async () => {
      const html = `
      <gux-input-search lang="en">
        <input name="search" type="search" placeholder="Enter a search" />
      </gux-input-search>`;
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-input-search');

      const clearButtonNoInput = await element.find('button.gux-clear-button');
      await a11yCheck(page, [], 'Input is empty');
      expect(clearButtonNoInput).toBeNull();

      const input = await element.find('input[type="search"]');
      await input.press('KeyT');
      await input.press('KeyE');
      await input.press('KeyS');
      await input.press('KeyT');
      await page.waitForChanges();

      const clearButtonInput = await element.find('button.gux-clear-button');
      await a11yCheck(page, [], 'Input contains a value');
      expect(clearButtonInput).not.toBeNull();

      const value = await input.getProperty('value');
      expect(value).toBe('test');
    });
  });
});
