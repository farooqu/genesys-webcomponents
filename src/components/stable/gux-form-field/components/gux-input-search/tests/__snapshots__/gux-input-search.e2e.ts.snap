// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-input-search #render should render component as expected (1) 1`] = `"<!----><div class=\\"gux-input-container\\"><gux-icon class=\\"gux-search-icon hydrated\\" icon-name=\\"search\\"></gux-icon> <input name=\\"search\\" type=\\"search\\" placeholder=\\"Enter a search\\"> </div>"`;

exports[`gux-input-search #render should render component as expected (2) 1`] = `"<!----><div class=\\"gux-input-container\\"><gux-icon class=\\"gux-search-icon hydrated\\" icon-name=\\"search\\"></gux-icon> <input name=\\"search\\" type=\\"search\\" placeholder=\\"Enter a search\\" value=\\"test\\"> <button class=\\"gux-clear-button\\" tabindex=\\"-1\\" type=\\"button\\" title=\\"Clear\\"><gux-icon icon-name=\\"close\\" class=\\"hydrated\\"></gux-icon></button></div>"`;

exports[`gux-input-search #render should render component as expected (3) 1`] = `"<!----><div class=\\"gux-input-container gux-disabled\\"><gux-icon class=\\"gux-search-icon hydrated\\" icon-name=\\"search\\"></gux-icon> <input name=\\"search\\" type=\\"search\\" placeholder=\\"Enter a search\\" disabled=\\"\\"> </div>"`;
